### PR TITLE
contrib: testgen: remove redundant base58 implementation

### PR DESCRIPTION
**Pull request by theStack**
This PR removes the redundant base58 implementation [contrib/testgen/base58.py](https://github.com/bitcoin/bitcoin/blob/master/contrib/testgen/base58.py) for the test generation script `gen_key_io_test_vectors.py` and uses the one from the test framework instead. Additionally, three other cleanups/improvements are done:
- import script operator constants `OP_*` from test framework instead of manually defining them
- add Python path to test framework directly in the script (via `sys.path.append(...)`) instead of needing the caller to specify `PYTHONPATH=...` on the command line (the same approach is done for the signet miner and the message capture scripts)
- rename `chars` to `b58chars` in the test_framework.address module (is more explicit and makes the diff for the base58 replacement smaller)

### CODE REVIEW

**Comment by laanwj** [comment]
Code review and lightly tested ACK 94c25d47539cf09b6dcb9e5211e5abb7c103725a
This is a non-controversial cleanup and makes the testgen commands easier to use.

File: contrib/testgen/gen_key_io_test_vectors.py [hunk 1]
```diff
@@ -6,16 +6,21 @@
 Generate valid and invalid base58/bech32(m) address and private key test vectors.
 
 Usage:
-    PYTHONPATH=../../test/functional/test_framework ./gen_key_io_test_vectors.py valid 70 > ../../src/test/data/key_io_valid.json
-    PYTHONPATH=../../test/functional/test_framework ./gen_key_io_test_vectors.py invalid 70 > ../../src/test/data/key_io_invalid.json
+    ./gen_key_io_test_vectors.py valid 70 > ../../src/test/data/key_io_valid.json
+    ./gen_key_io_test_vectors.py invalid 70 > ../../src/test/data/key_io_invalid.json
 '''
 # 2012 Wladimir J. van der Laan
```
Comment (laanwj) [inline contrib/testgen/gen_key_io_test_vectors.py:12]: Nit: could update this to the normal copyright header.

File: contrib/testgen/gen_key_io_test_vectors.py [hunk 2]
```diff
@@ -114,8 +109,10 @@ def is_valid(v):
     '''Check vector v for validity'''
     if len(set(v) - set(b58chars)) > 0:
         return is_valid_bech32(v)
-    result = b58decode_chk(v)
-    if result is None:
+    try:
+        payload, version = base58_to_byte(v)
+        result = bytes([version]) + payload
+    except AssertionError:  # thrown if checksum doesn't match
```
Comment (laanwj) [inline contrib/testgen/gen_key_io_test_vectors.py:115]: Catching an AssertionError for expected error handling is a bit strange. Normally one'd use `ValueError` or such for invalid values.

File: contrib/testgen/gen_key_io_test_vectors.py (hunk 1 above)
Comment (theStack) [reply contrib/testgen/gen_key_io_test_vectors.py]: Thanks, removed this line as well as the redundant MIT license information in the next line and updated the year in the header on top.

File: contrib/testgen/gen_key_io_test_vectors.py (hunk 2 above)
Comment (theStack) [reply contrib/testgen/gen_key_io_test_vectors.py]: Agreed that AssertionError is too generic for this case, I added a commit which changes this to `ValueError`, as suggested.

**Comment by laanwj** [comment]
Code review ACK 65c49ac750ba39801b349d0a59c27471dfa9868e

**Pull request merged**